chat-widget/src/hooks/useFileUploadProgress.ts

```diff
@@ -0,0 +1,79 @@
+import { useState, useCallback } from "react";
+
+export interface UploadProgressItem {
+    id: string;
+    progress: number;
+    status: "uploading" | "completed" | "error";
+}
+
+export const useFileUploadProgress = () => {
+    const [uploadProgress, setUploadProgress] = useState<Map<string, UploadProgressItem>>(new Map());
+    const [intervals, setIntervals] = useState<Map<string, NodeJS.Timeout>>(new Map());
+
+    const simulateUpload = useCallback((fileId: string) => {
```

Comment:
could you please help elaborate what exactly this component doing? why the uploading process is simulated? can it always be completed in 2.5 s?